fe1-web/src/features/lao/components/LaoItem.tsx

```diff
@@ -50,13 +51,12 @@ const LaoItem = ({ lao, isFirstItem, isLastItem }: IPropTypes) => {
         getNetworkManager().connect(address),
       );
 
-      const channel = getLaoChannel(lao.id.valueOf());
-      if (!channel) {
-        throw new Error('The given LAO ID is invalid');
-      }
-
-      // subscribe to the lao channel on the new connections
-      await subscribeToChannel(channel, connections);
+      // subscribe to the lao channel (or all previously subscribed to channels) on the new connection
+      await Promise.all(
+        lao.subscribed_channels.map((channel) =>
+          subscribeToChannel(lao.id, dispatch, channel, connections),
```

Comment:
I've skipped over it the first few times, waiting to see its usage, but I have to say that this new API does feel quite weird to me, as it leads to a lot of code being repeated here and there.
Can you walk me through why we need such a complex function for subscribe ?

In particular, I'm thinking as opposed to having simple APIs such as 
`connect(lao, dispatch)`
`resubscribe(lao, dispatch)`
that abstracts all of that away

Reply:
Yes of course :) The main problem is #1132. To fix it I wanted to keep track of all subscribed channels for each LAO and thus `subscribeToChannel` must be able to to change the redux state (and thus have access to `dispatch`). Analogously it needs to know for which lao we are doing the subscribe. Lastly, the `channel` argument should be self-explanatory.

Now `dispatch` could be removed or made an optional argument if the function would simply use the global `dispatch` function. Removing it would require us to always configure the global store in tests while as an optional argument we could still use dependency injection in tests. But since we always have access to a dispatch function where `subscribeToChannel`, I thought we can also make it a required argument. What do you think?

Also the last argument, `connections`, already is optional. This is for the case where you have multiple connections for the same LAO (e.g. organizer and witnesses) but only want to send a message on a specific connection. (For example the subscribe & catchup only on the newly added connection to the new witness)

Reply:
Makes sense, though I might have an idea for #1132 . I'll add it there